### ISSUE TITLE
Return of mixed in `$this->createQueryBuilder()->getQuery()->getResult()` when alias is `ev` or `event`

### ISSUE THREAD

**Issue by ruudk** [phpstan/phpstan-doctrine#608]
I have no idea what this is, or what causes it.

```php
 // This all works: list<Event>
dumpType($this->createQueryBuilder('e')->getQuery()->getResult());
dumpType($this->createQueryBuilder('my_name')->getQuery()->getResult());
dumpType($this->createQueryBuilder('oo')->getQuery()->getResult());
dumpType($this->createQueryBuilder('abcdef')->getQuery()->getResult());
dumpType($this->createQueryBuilder('user')->getQuery()->getResult());
dumpType($this->createQueryBuilder('av')->getQuery()->getResult());
dumpType($this->createQueryBuilder('eve')->getQuery()->getResult());
dumpType($this->createQueryBuilder('even')->getQuery()->getResult());

// This returns: mixed
dumpType($this->createQueryBuilder('ev')->getQuery()->getResult());
dumpType($this->createQueryBuilder('event')->getQuery()->getResult());
```

I also tried this on a different entity repository, for an entity named differently: Type. There, it gives the same problem.

So what is the cause for `ev` and `event` to behave weird?

**Comment by janedbal** [comment]
Replicated: https://github.com/phpstan/phpstan-doctrine/pull/609

**Comment by ruudk** [comment]
I was just using Xdebug to find the issue:

It's throwing here: 
https://github.com/phpstan/phpstan-doctrine/blob/be66017631e5289e8a3bd8973e59781642da40fd/src/Type/Doctrine/ArgumentsProcessor.php#L61-L66

**Comment by ruudk** [comment]
It seems that `$className` is the alias `ev` now:
![Screenshot 2024-09-12 at 16 39 15@2x](https://github.com/user-attachments/assets/0e3527d4-eff6-4561-a173-cefe7bf445bd)

**Comment by ruudk** [comment]
Interesting:
![Screenshot 2024-09-12 at 16 40 16@2x](https://github.com/user-attachments/assets/20df0c6c-12d1-4b41-ae6c-684b08ca2bfa)
It thinks this ConstantStringType is a ClassStringType

**Comment by ruudk** [comment]
It seems that I have a class called `ev` and `event`. The last one makes sense, somewhere. But the first one? Interesting.

https://github.com/phpstan/phpstan-src/blob/5480d27383ffee399037c31d461c5a85344128ca/src/Type/Constant/ConstantStringType.php#L83-L92

**Comment by janedbal** [comment]
Yes, because reflector see Event class as existing due to this stub:

- phar:///app/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/SourceLocator/SourceStubber/../../../../../jetbrains/phpstorm-stubs/event/event.stub

![image](https://github.com/user-attachments/assets/b4d72bb9-97b6-4eb9-845d-ee5a3e14faf1)

**Comment by janedbal** [comment]
@ondrejmirtes added this condition [here](https://github.com/phpstan/phpstan-doctrine/commit/0edf5b0cf1ff0a85adeefe4b5c1a6ba238f961cc). I think we should be more careful where we do this check. Imo only in `->from(class-name)` etc. Not in such a generic way for every argument.

**Comment by ondrejmirtes** [comment]
Yeah, it should throw only in arguments that expect class-string. Which is reasonable to assume in Doctrine that it must not be transient.

**Comment by janedbal** [comment]
I believe it would be the same issue as https://github.com/phpstan/phpstan-doctrine/issues/566

**Comment by janedbal** [comment]
Should be fixed with https://github.com/phpstan/phpstan-doctrine/pull/717 (since 2.0.15)

**Comment by ruudk** [comment]
Thank you ❤️
